x/cdp/keeper/auctions.go

```diff
@@ -0,0 +1,233 @@
+	debtChange = sdk.ZeroInt()
+	collateralChange = sdk.ZeroInt()
+	depositAmount := dep.Amount[0].Amount
+	depositDenom := dep.Amount[0].Denom
+	for depositAmount.GTE(auctionSize) {
+		// figure out how much debt is covered by one lots worth of collateral
+		depositDebtAmount := (sdk.NewDecFromInt(auctionSize).Quo(sdk.NewDecFromInt(totalCollateral))).Mul(sdk.NewDecFromInt(debt)).RoundInt()
+		// start an auction for one lot, attempting to raise depositDebtAmount
+		_, err := k.auctionKeeper.StartCollateralAuction(
+			ctx, types.LiquidatorMacc, sdk.NewCoin(depositDenom, auctionSize), sdk.NewCoin(principalDenom, depositDebtAmount), []sdk.AccAddress{dep.Depositor},
+			[]sdk.Int{auctionSize}, sdk.NewCoin(k.GetDebtDenom(ctx), depositDebtAmount))
+		if err != nil {
+			return sdk.ZeroInt(), sdk.ZeroInt(), err
+		}
+		depositAmount = depositAmount.Sub(auctionSize)
+		totalCollateral = totalCollateral.Sub(auctionSize)
+		debt = debt.Sub(depositDebtAmount)
+		// subtract one lot's worth of debt from the total debt covered by this deposit
+		debtChange = debtChange.Add(depositDebtAmount)
+		collateralChange = collateralChange.Add(auctionSize)
+
+	}
+	return debtChange, collateralChange, nil
+}
+
+// CreateAuctionFromPartialDeposits creates an auction from the input partial deposits
+func (k Keeper) CreateAuctionFromPartialDeposits(ctx sdk.Context, partialDeps partialDeposits, debt sdk.Int, collateral sdk.Int, auctionSize sdk.Int, bidDenom string) (debtChange, collateralChange sdk.Int, err sdk.Error) {
+
+	returnAddrs := []sdk.AccAddress{}
+	returnWeights := []sdk.Int{}
+	for _, pd := range partialDeps {
+		returnAddrs = append(returnAddrs, pd.Depositor)
+		returnWeights = append(returnWeights, pd.DebtShare)
+	}
+	_, err = k.auctionKeeper.StartCollateralAuction(ctx, types.LiquidatorMacc, sdk.NewCoin(partialDeps[0].Amount[0].Denom, auctionSize), sdk.NewCoin(bidDenom, partialDeps.SumDebt()), returnAddrs, returnWeights, sdk.NewCoin(k.GetDebtDenom(ctx), partialDeps.SumDebt()))
+	if err != nil {
+		return sdk.ZeroInt(), sdk.ZeroInt(), err
+	}
+	debtChange = partialDeps.SumDebt()
+	collateralChange = partialDeps.SumCollateral()
+	return debtChange, collateralChange, nil
+}
+
+// NetSurplusAndDebt burns surplus and debt coins equal to the minimum of surplus and debt balances held by the liquidator module account
+// for example, if there is 1000 debt and 100 surplus, 100 surplus and 100 debt are burned, netting to 900 debt
+func (k Keeper) NetSurplusAndDebt(ctx sdk.Context) sdk.Error {
+	totalSurplus := k.GetTotalSurplus(ctx, types.LiquidatorMacc)
+	debt := k.GetTotalDebt(ctx, types.LiquidatorMacc)
+	netAmount := sdk.MinInt(totalSurplus, debt)
+	if netAmount.GT(sdk.ZeroInt()) {
+		surplusToBurn := netAmount
+		err := k.supplyKeeper.BurnCoins(ctx, types.LiquidatorMacc, sdk.NewCoins(sdk.NewCoin(k.GetDebtDenom(ctx), netAmount)))
+		if err != nil {
+			return err
+		}
```

Comment:
This if could be
```go
if netAmount.IsZero() {
    return nil
}
surplusToBurn := ...
```
It would remove a level of indentation from the subsequent section.